### ISSUE TITLE
Remove umfpack dependency from random walker segmentation.

### ISSUE THREAD

**Issue by r3vl1s** [scikit-image/scikit-image#1362]
This is a low priority issue as it doesn't affect random walker functionality, but as of scipy 0.14 umfpack is deprecated (see http://docs.scipy.org/doc/scipy-dev/reference/release.0.14.0.html#scipy-sparse-linalg).

It's used to speed up some cases in random_walker_segmentation.py from skimage.segmentation.
There seems to be an unmaintained scikit (https://github.com/rc/scikit-umfpack). 

I think removing it from random_walker_segmentation.py is a good idea.

**Comment by jni** [comment]
I concur. @JDWarner? Is this a big loss?

**Comment by JDWarner** [comment]
pyamg is where the real efficiency improvement lies, in the testing I've done. That said @emmanuelle is the most familiar with these branching pathways.

**Comment by emmanuelle** [comment]
When I wrote a first version of the random walker code, depending on the way that scipy had been built, libumfpack was not always used for linear algebra operations. And some operations (especially the conjugate gradient) were very slow without umfpack. But I think that now most people have a scipy that was built in a "standard way" and won't encounter this problem. We could probably get rid of this piece code quite safely.

On the other hand, as @JDWarner is mentioning, the true speed gain is when using pyamg, with the multigrid computation. However pyamg is not very much maintained these days, so it's not great to have it as a dependency...

**Comment by JDWarner** [comment]
@emmanuelle we could roll our own iterative solver and ditch all of this. I know that's an option you floated a while back... decent amount of work, but might be for the best long-term.

**Comment by emmanuelle** [comment]
Yes, it's easy to replace the linear system inversion by a fixed-point equation that is a set of local updates. The issue is that when doing so, one loses the multigrid acceleration. of pyamg For example, if there are only a few markers in a large image, it will take a large number of iterations to broadcast the information of the markers to the other pixels. I think pyamg uses coarser aggregated versions of the graph of pixels, resulting in a faster convergence.

I could open a pull request with the code that I already have (using the local fixed-point equations) so that we can discuss such matters.

**Comment by soupault** [comment]
@emmanuelle 

> I could open a pull request with the code that I already have (using the local fixed-point equations) so that we can discuss such matters.

Please, go for it. :)
